### DISCUSSION TITLE
Allow specific prefix for command line generation

### DISCUSSION THREAD

**Discussion by qortex** [cyrilletuzi/vscode-angular-schematics#610] [Ideas]
In my `nx` setting, I use `nx g component ...` or even `yarn ng g component ...`.

The extension seems to use only `ng g component`.

Would it be possible to have a setting where the `ng` part could be customized?

**Comment by cyrilletuzi** [comment]
Custom configurations (Nx, yarn,...) support may come in an enhanced version, which I plan to develop this summer.

2 questions:
- does it change something compared to `ng g` or is it just cosmetic? if so, what does it change exactly?
- is the tool used specified in some JSON setting file? if so, which one and where?

**Reply by qortex** [reply]
1. I'm not entirely sure (I just start with using Nx), but as far as I understand and as generators are concerned, `nx` just calls `ng` under the hood and sometimes customizes a few options on the way. So arguments are the same (I basically copy/paste the command created by the extension and replace the first `ng` with `nx`).
2. As a result, I think the specification for the command lines are the same.

**Reply by qortex** [reply]
Looks like a few settings are added and are general to all generators: https://nx.dev/latest/angular/cli/generate

For example the `--dryRun` toggle, not sure it is present in `ng` setting?

**Reply by qortex** [reply]
Actually no, the extension generates `ng g @schematics/angular:component ...` and I use `nx g component ...`.

**Comment by rmaugusto** [comment]
In my case I dont use global ng and I prefer to use ./node_modules/@angular/cli/bin/ng to follow project version.
Add a prefix to command would be great !

**Comment by cyrilletuzi** [comment]
Hi @qortex, @rmaugusto and @rene-ott, for information, an extended version of the extension is now available: **Schematics Pro**. You will find all the details on the [dedicated website](https://www.cyrilletuzi.com/schematics-pro/).

Regarding this discussion, it solves 2 things:
- for classic schematics, global Angular CLI is not required anymore as Schematics Pro has its own schematics generator; it also has the benefit to be faster and to avoid other issues (like shell issues, or finding dependencies not in `node_modules`)
- for third-party schematics (like the Nx ones), Schematics Pro takes into account your package manager to launch the commands (npx for npm, yarn, pnpm...), so global Angular CLI is not required anymore in this case too.

Concerning Nx specific features, a future Schematics Pro update will be dedicated to Nx.

**Comment by cyrilletuzi** [comment]
Latest version (v6) of the extension has its own generator and does not rely on Angular CLI anymore.
